gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -637,7 +637,7 @@ public JsonWriter value(Number value) throws IOException {
     out.close();
 
     int size = stackSize;
-    if (size > 1 || size == 1 && stack[size - 1] != NONEMPTY_DOCUMENT) {
+    if ((size > 1 || size == 1) && stack[size - 1] != NONEMPTY_DOCUMENT) {
```

Comment:
This doesn't look right. The existing code is equivalent to
```
if (size > 1 || (size == 1 && stack[size - 1] != NONEMPTY_DOCUMENT)) {
```
Parenthesizing it like that would be helpful, but this version changes the meaning.